cmd/earthly/secret_cmds.go

```diff
@@ -276,17 +285,54 @@ func (app *earthlyApp) fullSecretPath(path string) (string, error) {
 		return path, nil
 	}
 
-	if app.orgName == "" {
-		return "", fmt.Errorf("the --org flag is required")
+	orgName, projectName, isPersonal, err := app.getOrgAndProject(ctx, cloudClient)
+	if err != nil {
+		return "", err
 	}
-	if app.projectName == "" {
-		return "", fmt.Errorf("the --project flag is required")
+
+	if isPersonal && projectName == "" && !strings.HasPrefix(path, "/user") {
+		if path == "/" {
+			return "/user", nil
+		} else {
+			return fmt.Sprintf("/user%s", path), nil
+		}
 	}
 
 	// TODO: These values will eventually come from the new PROJECT command (if
-	// one is present). For now, we can use the flag/env values as a temporary
-	// measure.
-	return fmt.Sprintf("/%s/%s%s", app.orgName, app.projectName, path), nil
+	//   one is present). For now, we can use the flag/env values as a temporary
+	//   measure.
+	return fmt.Sprintf("/%s/%s%s", orgName, projectName, path), nil
+}
+
+func (app *earthlyApp) getOrgAndProject(ctx context.Context, client *cloud.Client) (org, project string, isPersonal bool, err error) {
+	if app.orgName != "" {
+		org = app.orgName
+	} else if app.cfg.Global.Org != "" {
+		org = app.cfg.Global.Org
+	}
```

Comment:
Similar to the other declare/reassign block, this could use a helper. `app.OrgName()` or `app.org()` or something would work.